spec/requests/shopping_list_items_spec.rb

```diff
@@ -224,4 +224,194 @@
+          'email' => user.email,
+          'name' => user.name
+        }
+      end
+      
+      let(:validator) { instance_double(GoogleIDToken::Validator, check: validation_data) }
+      
+      before do
+        allow(GoogleIDToken::Validator).to receive(:new).and_return(validator)
+      end
+      
+      context 'when all goes well' do
+        let(:params) { { shopping_list_item: { quantity: 5, notes: 'To make locks' } } }
+        let(:list_item) { shopping_list.list_items.first }
+
+        before do
+          second_list = create(:shopping_list, user: user, master_list: master_list)
+          second_list.list_items.create!(description: list_item.description, quantity: 2)
+          master_list.add_item_from_child_list(shopping_list.list_items.first)
+          master_list.add_item_from_child_list(shopping_list.list_items.last) # for the sake of realism
+          master_list.add_item_from_child_list(second_list.list_items.first) 
+        end
+
+        it 'updates the regular list item' do
+          update_item
+          expect(list_item.reload.attributes).to include(
+                                                          'quantity' => 5,
+                                                          'notes' => 'To make locks'
+                                                        )
+        end
+
+        it 'updates the item on the master list' do
+          update_item
+          expect(master_list.list_items.first.attributes).to include(
+                                                                      'description' => list_item.description,
+                                                                      'quantity' => 7,
+                                                                      'notes' => 'To make locks'
+                                                                    )
+        end
+
+        it 'updates the regular list', :aggregate_failures do
+          t = Time.now + 3.days
+
+          Timecop.freeze(t) do
+            update_item
+            # use `be_within` even though the time will be set to the time Timecop
+            # has frozen because Rails (Postgres?) sets the last three digits of
+            # the timestamp to 0, which was breaking stuff in CI (but somehow not
+            # in dev).
+            expect(shopping_list.reload.updated_at).to be_within(0.05.seconds).of(t)
+            expect(master_list.reload.updated_at).not_to be_within(0.05.seconds).of(t)
+          end
+        end
+
+        it 'returns the master list item and the regular list item', :aggregate_failures do
+          update_item
+
+          # The serialization isn't as simple as .to_json and it makes it hard to determine if the JSON
+          # string is correct as some attributes are out of order and the timestamps are serialized
+          # differently. Here we grab the individual items and then we'll filter out the timestamps to
+          # verify them.
```

Comment:
The values of the timestamps, to the extent they matter, are covered in other specs.